aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -46,32 +51,36 @@ const UNSIGNED_PAYLOAD: &str = "UNSIGNED-PAYLOAD";
 const STREAMING_UNSIGNED_PAYLOAD_TRAILER: &str = "STREAMING-UNSIGNED-PAYLOAD-TRAILER";
 
 #[derive(Debug, PartialEq)]
-pub(super) struct HeaderValues<'a> {
-    pub(super) content_sha256: Cow<'a, str>,
-    pub(super) date_time: String,
-    pub(super) security_token: Option<&'a str>,
-    pub(super) signed_headers: SignedHeaders,
+pub(crate) struct HeaderValues<'a> {
+    pub(crate) content_sha256: Cow<'a, str>,
+    pub(crate) date_time: String,
+    pub(crate) security_token: Option<&'a str>,
+    pub(crate) signed_headers: SignedHeaders,
+    #[cfg(feature = "sigv4a")]
+    pub(crate) region_set: Option<&'a str>,
 }
 
 #[derive(Debug, PartialEq)]
-pub(super) struct QueryParamValues<'a> {
-    pub(super) algorithm: &'static str,
-    pub(super) content_sha256: Cow<'a, str>,
-    pub(super) credential: String,
-    pub(super) date_time: String,
-    pub(super) expires: String,
-    pub(super) security_token: Option<&'a str>,
-    pub(super) signed_headers: SignedHeaders,
+pub(crate) struct QueryParamValues<'a> {
+    pub(crate) algorithm: &'static str,
+    pub(crate) content_sha256: Cow<'a, str>,
+    pub(crate) credential: String,
+    pub(crate) date_time: String,
+    pub(crate) expires: String,
+    pub(crate) security_token: Option<&'a str>,
+    pub(crate) signed_headers: SignedHeaders,
+    #[cfg(feature = "sigv4a")]
```

Comment:
(and in a case like this, is there any harm in having this field exist when sigv4a isn't enabled?)

Reply:
Only a few bytes of memory. I'll remove the gates.

Reply:
never mind, removing them would be a huge pain.